lib/core/proxy.c

```diff
@@ -733,7 +743,11 @@ static int on_1xx(h2o_httpclient_t *client, int version, int status, h2o_iovec_t
             h2o_push_path_in_link_header(self->src_req, headers[i].value.base, headers[i].value.len);
     }
 
-    if (status != 101) {
+    if (status == 101) {
+        /* switching protocol */
+    } else if (status == 100) {
+        /* we don't need to forward 100 since protocol handlers have already done */
```

Comment:
Considering that lib/common/http1client.c consumes and reacts to 100, do we need to have the signal propagated to proxy.c?

Reply:
Addressed in https://github.com/h2o/h2o/pull/3316/commits/0ce40027978ba0b383a8095c44ad55f4e0e1b98c

Reply:
Ah sorry for the lack of explanation, yeah, considering your comments I thought that consuming here is consistent and can simplify the things. Thanks!